src/odemis/acq/stream/_projection.py

```diff
@@ -385,12 +387,20 @@ def _project2Polar(self, pos):
         # still running this method, the dict might get new entries again, though it should be empty.
         polar_cache = self._polar_cache
 
-        if pos in polar_cache:
-            polar_data = polar_cache[pos]
+        # if ebeam_pos + (pol_pos,) in polar_cache:
```

Comment:
That looks fine. Just remove the old commented-out lines.

BTW, instead of the `if`, you could use `try/except KeyError`. It avoids having a long `if` condition.